### PR TITLE
Allow disabling torch 2_0 attention

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
A little tweak so that people can easily switch off PT 2.0 attention if needed

**Comment by patrickvonplaten** [comment]
Now the PT attention can be switched off for the VAE by doing:

`vae.decoder.mid_block.attentions[0]._use_2_0_attn = False` 

cc @echarlaix

File: src/diffusers/models/attention.py [hunk 1]
```diff
@@ -142,9 +143,8 @@ def forward(self, hidden_states):
 
         scale = 1 / math.sqrt(self.channels / self.num_heads)
 
-        use_torch_2_0_attn = (
-            hasattr(F, "scaled_dot_product_attention") and not self._use_memory_efficient_attention_xformers
-        )
+        _use_2_0_attn = self._use_2_0_attn and not self._use_memory_efficient_attention_xformers and self._use_2_0_attn
```
Comment (sayakpaul) [inline src/diffusers/models/attention.py:146]: A bit long to read, but fine if you are fine.

Comment (patrickvonplaten) [inline src/diffusers/models/attention.py:146]:
```suggestion
        _use_2_0_attn = self._use_2_0_attn and not self._use_memory_efficient_attention_xformers
```

**Review by pcuenca — APPROVED** [review]

**Pull request merged**